code/modules/modular_computers/computers/machinery/console_presets.dm

```diff
@@ -19,96 +20,77 @@
 		cpu.battery_module = new/obj/item/weapon/computer_hardware/battery_module/super(cpu)
 	if(_has_aislot)
 		cpu.ai_slot = new/obj/item/weapon/computer_hardware/ai_slot(cpu)
-	if(_software_locked)
-		cpu.software_locked = 1
-	install_programs()
+	if(_enrolled)
+		cpu.enrolled = _enrolled
+	install_programs(get_preset_programs(_app_preset_name))
 
-// Override in child types to install preset-specific programs.
-/obj/machinery/modular_computer/console/preset/proc/install_programs()
-	return
+/obj/machinery/modular_computer/console/preset/proc/get_preset_programs(var/app_preset_name)
+	for (var/datum/modular_computer_app_presets/prs in ntnet_global.available_software_presets)
+		//var/datum/modular_computer_app_presets/prs = new path()
+		if(prs.name == app_preset_name && prs.available == 1)
+			return prs.return_install_programs()
 
+// Override in child types to install preset-specific programs.
+/obj/machinery/modular_computer/console/preset/proc/install_programs(var/list/programs)
+	for (var/datum/computer_file/program/prog in programs)
+		cpu.hard_drive.store_file(prog)
 
 // ===== ADMIN CONSOLE =====
-/obj/machinery/modular_computer/console/preset/admin
+/obj/machinery/modular_computer/console/preset/all
 	console_department = "CC ITIQ"
-	desc = "A computer from the CC ITIQ Department. Only available on the odin"
+	desc = "A computer from the CC ITIQ Department. Only available on the Odin"
+	_app_preset_name = "all"
+	_enrolled = 1
 
 /obj/machinery/modular_computer/console/preset/admin/install_programs()
-	for(var/F in typesof(/datum/computer_file/program))
-		var/datum/computer_file/program/prog = new F
-		cpu.hard_drive.store_file(prog)
+	..()
 	cpu.battery_module = new/obj/item/weapon/computer_hardware/battery_module/lambda(cpu)
 
+// ===== Empty Test console CONSOLE =====
+/obj/machinery/modular_computer/console/preset/empty
+	 desc = "A stationary computer."
+
 // ===== ENGINEERING CONSOLE =====
 /obj/machinery/modular_computer/console/preset/engineering
```

Comment:
CE is supposed to have the lighting control thing too right? Good thing to make a CE subclass for